src/fat32.c

```diff
@@ -163,16 +163,27 @@ fat32_make_efi(uint32_t image_fd, uint32_t first_sector, uint32_t num_sectors,
   image_t img = {
     .fd = image_fd,
     .first_sector = first_sector,
-    .sectors_per_cluster = 16,
+    // TODO: The sectors_per_cluster should probably be computed automatically to ensure
+    // 1. There are at least 65525 clusters (to be considered fat32)
+    // 2. A cluster has at most 32768 byte
+    .sectors_per_cluster = 2,
```

Comment:
We could also simply set it to `1`, which is the minimum value. If that doesn't give us enough clusters then we can suggest the user (or enforce) a larger number of sectors.